test/integration/Client/client_lib_thread_interdependency_test.cpp

```diff
@@ -0,0 +1,499 @@
+            check_story_destroyed(tid, ret);
+            break;
+        case ThreadState::CHRONICLE_DESTROYED:
+            check_chronicle_destroyed(tid, ret);
+            break;
+        case ThreadState::THREAD_FINALIZED:
+            check_thread_finalized(tid, ret);
+            break;
+        case ThreadState::UNKNOWN:
+        default:
+            break;
+    }
+    //std::cout << "Thread " << tid << " checked its state: "
+    //          << get_state_name(static_cast<ThreadState>(shared_state[tid])) << std::endl;
+}
+
+void thread_body(struct thread_arg*t)
+{
+    // Thread Initialized
+    handle_return_value(t->tid, 0, ThreadState::THREAD_INITIALIZED);
+
+
+    // Chronicle Variables
+    std::string chronicle_name = (t->tid % 2 == 0) ? "CHRONICLE_2" : "CHRONICLE_1";//"CHRONICLE";
+    std::map <std::string, std::string> chronicle_attrs;
+    chronicle_attrs.emplace("Priority", "High");
+    int flags = 1;
+    // Chronicle creation
+    int ret = client->CreateChronicle(chronicle_name, chronicle_attrs, flags);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Chronicle created: tid={}, Ret: {}", t->tid, ret);
+    handle_return_value(t->tid, ret, ThreadState::CHRONICLE_CREATED);
+
+
+    // Story Variables
+    std::string story_name = (t->tid % 4 < 2) ? "STORY_1" : "STORY_2";
+    std::map <std::string, std::string> story_attrs;
+    flags = 2;
+    // Acquire story
+    auto acquire_ret = client->AcquireStory(chronicle_name, story_name, story_attrs, flags);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Story acquired: tid={}, Ret: {}", t->tid, acquire_ret.first);
+    handle_return_value(t->tid, acquire_ret.first, ThreadState::STORY_ACQUIRED);
+
+
+    // Release the story
+    ret = client->ReleaseStory(chronicle_name, story_name);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Story released: tid={}, Ret: {}", t->tid, ret);
+    handle_return_value(t->tid, ret, ThreadState::STORY_RELEASED);
+
+
+    // Destroy the story
+    ret = client->DestroyStory(chronicle_name, story_name);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Story destroyed: tid={}, Ret: {}", t->tid, ret);
+    handle_return_value(t->tid, ret, ThreadState::STORY_DESTROYED);
+
+
+    // Destroy the chronicle
+    ret = client->DestroyChronicle(chronicle_name);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Chronicle destroyed: tid={}, Ret: {}", t->tid, ret);
+    handle_return_value(t->tid, ret, ThreadState::CHRONICLE_DESTROYED);
+
+
+    // Thread Finalized
+    handle_return_value(t->tid, 0, ThreadState::THREAD_FINALIZED);
+}
+
+
+int main(int argc, char**argv)
+{
+    // Configuration
+    std::string conf_file_path = parse_conf_path_arg(argc, argv);
+    if(conf_file_path.empty())
+    {
+        std::exit(EXIT_FAILURE);
+    }
+    ChronoLog::ConfigurationManager confManager(conf_file_path);
+    int result = chronolog::chrono_monitor::initialize(confManager.CLIENT_CONF.CLIENT_LOG_CONF.LOGTYPE
```

Comment:
To me, it is more convenient to get the result directly from the console output especially when users are more likely to use an existing conf file generated from the deployment script where the client log file is already used by the `mutli_storyteller` test case. We can print the final results (pass/fail) to the screen and log the details in a separate log file different from the one that the deployment script uses.

Reply:
I think this is not a problem anymore as we separated the deployement from the client execution.